src/components/with-model.jsx

```diff
@@ -21,10 +21,13 @@ function withModel(WrappedComponent, createModel) {
       this.setState = () => {};
     }
 
-    async updateModel() {
+    async updateModel(firstTime) {
       const { app } = this.props;
       try {
         const model = await createModel(app, this.props);
+        if (firstTime && updateOnAppInvalidation) {
+          app.on('changed', this.updateModel);
```

Comment:
We need to remove the listener on unmount/destroy too (`app.removeListener('changed', fn)`)

Reply:
thanks!